src/bundle-manifest.ts

```diff
@@ -201,16 +201,24 @@ export function generateSharedDepsMergeStrategy(minEntrypoints?: number):
  * bundle for an application shell.
  */
 export function generateShellMergeStrategy(
-    shell: UrlString, minEntrypoints: number): BundleStrategy {
+    shell: UrlString, minEntrypoints?: number): BundleStrategy {
   return (bundles: Bundle[]): Bundle[] => {
+    if (minEntrypoints === undefined) {
+      minEntrypoints = 2;
+    }
     const newBundles = generateSharedDepsMergeStrategy(minEntrypoints)(bundles);
-    const shellBundle = newBundles.find((bundle) => bundle.files.has(shell));
-    const sharedBundle =
-        newBundles.find((bundle) => bundle.entrypoints.size > 1);
-    if (shellBundle && sharedBundle && shellBundle !== sharedBundle) {
+    let shellBundle = newBundles.find(bundle => bundle.files.has(shell));
+    if (!shellBundle) {
+      throw new Error(`No bundle found containing specified shell ${shell}`);
+    }
+    const sharedBundles = newBundles.filter(
+        (bundle) => bundle.entrypoints.size >= minEntrypoints &&
+            bundle !== shellBundle);
+    for (const sharedBundle of sharedBundles) {
       newBundles.splice(newBundles.indexOf(shellBundle), 1);
       newBundles.splice(newBundles.indexOf(sharedBundle), 1);
-      newBundles.push(mergeBundles([shellBundle, sharedBundle]));
+      shellBundle = mergeBundles([shellBundle, sharedBundle]);
```

Comment:
Nothing explicitly honors or disregards lazy/non-lazy bundles in any strategy code AFAIK.  Lazy bundles + shell and shared merge strategy generators are worth looking into but way beyond the scope of this.

The origin of this PR is actually just the result of noticing that finding the sharedBundle with .find() had the potential of fetching the wrong bundle if the shell was explicitly linked to by any entrypoints and based on arbitrary order of the bundles searched over.  I fixed the expectation of the behavior of sending in higher minEntrypoints values and lastly made the argument optional to make it compatible to the optional argument for generateSharedDepsMergeStrategy.  But fixing the find was the impetus.